src/clients/InventoryClient.ts

```diff
@@ -327,21 +359,166 @@ export class InventoryClient {
+   * Therefore, running balances can only be considered "excess" if the running balance as recorded in the
+   * PoolRebalanceLeaf is negative. This is denoting that the Hub is over-allocated on the Spoke.
+   * @param l1Token Token we are comparing running balances for against spoke pool targets.
+   * @param refundAmount Amount that will be refunded to the relayer. This value gets subtracted from running
+   * balance excesses before comparing with the spoke pool target, since refunds are taken out of spoke pool balances.
+   * @returns Dictionary of excess percentages for each chain. The excess percentage is the percentage of the
+   * excess running balance over the spoke pool target balance. If the absolute running balance is 0, then
+   * the excess percentage is 0. If the target is 0, then the excess percentage is infinite.
+   */
+  _getExcessRunningBalancePcts(
+    excessRunningBalances: { [chainId: number]: BigNumber },
+    l1Token: string,
+    refundAmount: BigNumber
+  ): { [chainId: number]: BigNumber } {
+    const pcts = Object.fromEntries(
+      Object.entries(excessRunningBalances).map(([chainId, excess]) => {
+        const target = this.hubPoolClient.configStoreClient.getSpokeTargetBalancesForBlock(
+          l1Token,
+          Number(chainId)
+        ).target;
+        const excessPostRelay = excess.sub(refundAmount);
+        const returnObj = {
+          pct: toBN(0),
+          target,
+          excess,
+          excessPostRelay,
+        };
+        // If target is greater than excess running balance, then pct will
+        // be set to 0. If target is 0 then pct is infinite.
+        if (target.lte(0)) {
+          returnObj.pct = toBN(MAX_UINT_VAL);
+        } else if (excessPostRelay.gt(target)) {
+          returnObj.pct = excessPostRelay.sub(target).mul(this.scalar).div(target);
+        }
+        return [chainId, returnObj];
+      })
+    );
+    this.log(`Computed excess running balances for ${l1Token}`, {
+      refundAmount,
+      excessRunningBalancePcts: Object.fromEntries(
+        Object.entries(pcts).map(([k, v]) => [
+          k,
+          {
+            ...pcts[k],
+            pct: formatFeePct(v.pct) + "%",
+          },
+        ])
+      ),
+    });
+    return Object.fromEntries(Object.entries(pcts).map(([k, v]) => [k, v.pct]));
+  }
+
+  async getExcessRunningBalancePcts(
+    l1Token: string,
+    refundAmount: BigNumber
+  ): Promise<{ [chainId: number]: BigNumber }> {
+    if (!isDefined(this.excessRunningBalancePromises[l1Token])) {
+      // @dev Save this as a promise so that other parallel calls to this function don't make the same call.
+      this.excessRunningBalancePromises[l1Token] = this.getLatestRunningBalances(l1Token);
+    }
+    const excessRunningBalances = lodash.cloneDeep(await this.excessRunningBalancePromises[l1Token]);
+    return await this._getExcessRunningBalancePcts(excessRunningBalances, l1Token, refundAmount);
```

Comment:
nit: I don't think this function returns a promise or is async
```suggestion
    return this._getExcessRunningBalancePcts(excessRunningBalances, l1Token, refundAmount);
```